packages/core/src/components/pagination/pagination.tsx

```diff
@@ -49,30 +77,105 @@ export class Pagination implements ComponentInterface {
     const pages = Array.from({ length: this.totalPages }, (_value, i) => i + 1);
     const currentPageOutOfBounds = currentPage < pages[0] || currentPage > pages[pages.length - 1];
 
+    const pageRange = this.getPageRange(pages, currentPage, this.totalPages, this.elementPositions);
+
     return (
-      <ul class="pagination">
-        <li class={(currentPage <= pages[0] || currentPageOutOfBounds) ? 'dso-page-hidden' : undefined}>
-          <a href={this.formatHref(pages[0])} aria-label="Vorige" onClick={e => currentPage && this.clickHandler(e, pages[currentPage - 2])}>
-            <dso-icon icon="chevron-left"></dso-icon>
-          </a>
-        </li>
-        {pages.map(page => (
-          <li key={page} class={currentPage === page ? 'active' : undefined}>
-            {currentPage === page
-              ? (
-                <span aria-current="page">{page}</span>
-              )
-              : (
-                <a href={this.formatHref(page)} onClick={e => this.clickHandler(e, page)}>{page}</a>
-              )}
+      <dso-responsive-element class="dso-pagination">
+        <ul class="pagination">
+          <li class={(currentPage <= pages[0] || currentPageOutOfBounds) ? 'dso-page-hidden' : undefined}>
+            <a href={this.formatHref(pages[0])} aria-label="Vorige" onClick={e => currentPage && this.clickHandler(e, pages[currentPage - 2])}>
+              <dso-icon icon="chevron-left"></dso-icon>
+            </a>
+          </li>
+          {(pageRange).map(page => (
+            <>
+              {(pages.indexOf(page) === pages.length - 1 && pages.length > (this.elementPositions - 2)) &&
+                <>
+                { (
+                    (pages.indexOf(currentPage) === 0 && this.elementPositions === 6) ||
```

Comment:
misschien kunnen we deze conditions beter vastleggen in een eigen functie, dan wordt de code leesbaarder omdat de functienaam documenterend zal zijn.

Daarnaast is dit nu lastig onderhoudbaar, het aanpassen van `sizePositionsMap` is niet voldoende om de beschikbare posities aan te passen, ik denk dat we de controle volledig in die map willen vastleggen.

Reply:
Ik heb geprobeerd het wat dynamischer te maken.